src/shady-css/parser.js

```diff
@@ -225,22 +229,21 @@ class Parser {
       }
     }
 
-    rule = tokenizer.slice(ruleStart, ruleEnd);
-
     // A ruleset never contains or ends with a semi-colon.
     if (tokenizer.currentToken.is(Token.type.propertyBoundary)) {
-      let colonIndex = rule.indexOf(':');
+      let declarationName = tokenizer.slice(ruleStart, colon.previous);
```

Comment:
Yeah, I mentioned at the top that this is just the first half of the change needed to resolve #3. The second half will be to do exactly what you described.